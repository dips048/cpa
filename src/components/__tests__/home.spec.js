import React from "react"
import renderer from "react-test-renderer"
import Home from "../home"
import { render, fireEvent } from "@testing-library/react"
import TestRenderer from "react-test-renderer"
import ContactUsModel from "../contact-us-modal"
import CallLink from "../callLink"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"

describe("Home", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Home />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("renders `FontAwesomeIcon`, `ContactUsModel`, `CallLink` components", () => {
    const testRenderer = TestRenderer.create(
      <Home />
    )
    const testInstance = testRenderer.root
    expect(testInstance.findAllByType(FontAwesomeIcon).length).toEqual(4)
    expect(testInstance.findByType(ContactUsModel)).toBeTruthy()
    expect(testInstance.findAllByType(CallLink).length).toEqual(1)
  })
  
  test("on clicking `Free Consultation` it should open Free Consultation modal", () => {
    const { getByText, getByRole } = render(
      <Home />
    )
    const freeConsultation = getByText("Free Consultation")
    expect(freeConsultation).toBeInTheDocument()
    fireEvent.click(freeConsultation);
    expect(getByRole("dialog")).toBeInTheDocument();
    const sendButton = getByText("Send");
    fireEvent.click(sendButton);
  })
})

