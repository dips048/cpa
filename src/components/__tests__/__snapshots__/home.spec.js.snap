// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <div
    className="d-md-flex container"
  >
    <div
      className="text-center order-md-2 mx-auto"
    />
    <div
      className="row order-md-1 py-2 flex-md-grow-1"
    >
      <div
        className="col-md-8 text-center d-flex flex-column justify-content-center"
        id="home-page-tagline"
      >
        <h1>
          Account On Us
        </h1>
        <p>
          We offer services for business owners, executives, and individuals.
        </p>
        <button
          className="btn btn-primary btn-lg"
          onClick={[Function]}
        >
          Free Consultation
        </button>
      </div>
    </div>
  </div>
  <div
    className="d-flex theme-color"
  >
    <div
      className="container"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <div
        className="row pt-5"
      >
        <div
          className="col-lg-12"
        >
          <h1>
            CPA Solutions, Accounting
          </h1>
          <p>
            If you are looking for a blend of personal service and expertise, you have come to the right place! We offer a broad range of services for business owners, executives and independent professionals. Our rates are affordable. We are experienced and friendly.
          </p>
        </div>
      </div>
      <div
        className="row my-5 mx-3 align-xs-center text-center text-sm-start d-flex default-color"
      >
        <div
          className="col-lg-6 col-md-12 my-4"
        >
          <div
            className="row"
          >
            <div
              className="col-sm-2 icon-color text-center"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-compass fa-4x "
                data-icon="compass"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 256C288 273.7 273.7 288 256 288C238.3 288 224 273.7 224 256C224 238.3 238.3 224 256 224C273.7 224 288 238.3 288 256zM0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM325.1 306.7L380.6 162.4C388.1 142.1 369 123.9 349.6 131.4L205.3 186.9C196.8 190.1 190.1 196.8 186.9 205.3L131.4 349.6C123.9 369 142.1 388.1 162.4 380.6L306.7 325.1C315.2 321.9 321.9 315.2 325.1 306.7V306.7z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
            <div
              className="col-sm-10"
            >
              <h3>
                Services For Individuals
              </h3>
              <p>
                You get one-on-one guidance that helps manage risk and improve performance.
                <a
                  href="/services/services-for-individuals"
                >
                  learn more
                </a>
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-lg-6 col-md-12 my-4"
        >
          <div
            className="row"
          >
            <div
              className="col-sm-2 icon-color text-center"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-briefcase fa-4x "
                data-icon="briefcase"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M320 336c0 8.844-7.156 16-16 16h-96C199.2 352 192 344.8 192 336V288H0v144C0 457.6 22.41 480 48 480h416c25.59 0 48-22.41 48-48V288h-192V336zM464 96H384V48C384 22.41 361.6 0 336 0h-160C150.4 0 128 22.41 128 48V96H48C22.41 96 0 118.4 0 144V256h512V144C512 118.4 489.6 96 464 96zM336 96h-160V48h160V96z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
            <div
              className="col-sm-10"
            >
              <h3>
                Business Services
              </h3>
              <p>
                We take care of your business for you, so you can get back to the job of running your business.
                <a
                  href="/services/business-services"
                >
                  learn more
                </a>
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-lg-6 col-md-12 my-4"
        >
          <div
            className="row"
          >
            <div
              className="col-sm-2 icon-color text-center"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-calculator fa-4x "
                data-icon="calculator"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M336 0h-288C22.38 0 0 22.38 0 48v416C0 489.6 22.38 512 48 512h288c25.62 0 48-22.38 48-48v-416C384 22.38 361.6 0 336 0zM64 208C64 199.2 71.2 192 80 192h32C120.8 192 128 199.2 128 208v32C128 248.8 120.8 256 112 256h-32C71.2 256 64 248.8 64 240V208zM64 304C64 295.2 71.2 288 80 288h32C120.8 288 128 295.2 128 304v32C128 344.8 120.8 352 112 352h-32C71.2 352 64 344.8 64 336V304zM224 432c0 8.801-7.199 16-16 16h-128C71.2 448 64 440.8 64 432v-32C64 391.2 71.2 384 80 384h128c8.801 0 16 7.199 16 16V432zM224 336c0 8.801-7.199 16-16 16h-32C167.2 352 160 344.8 160 336v-32C160 295.2 167.2 288 176 288h32C216.8 288 224 295.2 224 304V336zM224 240C224 248.8 216.8 256 208 256h-32C167.2 256 160 248.8 160 240v-32C160 199.2 167.2 192 176 192h32C216.8 192 224 199.2 224 208V240zM320 432c0 8.801-7.199 16-16 16h-32c-8.799 0-16-7.199-16-16v-32c0-8.801 7.201-16 16-16h32c8.801 0 16 7.199 16 16V432zM320 336c0 8.801-7.199 16-16 16h-32c-8.799 0-16-7.199-16-16v-32C256 295.2 263.2 288 272 288h32C312.8 288 320 295.2 320 304V336zM320 240C320 248.8 312.8 256 304 256h-32C263.2 256 256 248.8 256 240v-32C256 199.2 263.2 192 272 192h32C312.8 192 320 199.2 320 208V240zM320 144C320 152.8 312.8 160 304 160h-224C71.2 160 64 152.8 64 144v-64C64 71.2 71.2 64 80 64h224C312.8 64 320 71.2 320 80V144z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
            <div
              className="col-sm-10"
            >
              <h3>
                Tax Services
              </h3>
              <p>
                We pride ourselves on being very efficient, affordable, and of course, extremely discreet.
                <a
                  href="/services/tax-services"
                >
                  learn more
                </a>
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-lg-6 col-md-12 my-4"
        >
          <div
            className="row"
          >
            <div
              className="col-sm-2 icon-color text-center"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-book fa-4x "
                data-icon="book"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M448 336v-288C448 21.49 426.5 0 400 0H96C42.98 0 0 42.98 0 96v320c0 53.02 42.98 96 96 96h320c17.67 0 32-14.33 32-31.1c0-11.72-6.607-21.52-16-27.1v-81.36C441.8 362.8 448 350.2 448 336zM143.1 128h192C344.8 128 352 135.2 352 144C352 152.8 344.8 160 336 160H143.1C135.2 160 128 152.8 128 144C128 135.2 135.2 128 143.1 128zM143.1 192h192C344.8 192 352 199.2 352 208C352 216.8 344.8 224 336 224H143.1C135.2 224 128 216.8 128 208C128 199.2 135.2 192 143.1 192zM384 448H96c-17.67 0-32-14.33-32-32c0-17.67 14.33-32 32-32h288V448z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
            <div
              className="col-sm-10"
            >
              <h3>
                QuickBooks Services
              </h3>
              <p>
                QuickBooks is the ideal business accounting software for small to mid-sized business owners.
                <a
                  href="/services/services-for-quick-books"
                >
                  learn more
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
    style={
      Object {
        "zIndex": 1,
      }
    }
  >
    <div
      className="row my-5"
    >
      <div
        className="col-lg-12"
      >
        <h1
          className="h1 text-center"
        >
          Accounting is what we do best, you're busy enough with the rest
        </h1>
        <div
          className="col-lg-12 mb-4"
        >
          <hr />
        </div>
        <div
          className="col-12 d-flex row"
        >
          <div
            className="col-lg-5 order-lg-2"
          >
            <div
              className="coffee"
            />
          </div>
          <div
            className="col-lg-6 order-lg-1"
          >
            <p
              className="lead"
            >
              CPA Solutions is licensed in VT and specializes in Accounting. We offer a broad range of services for business owners, executives, and independent professionals. We are professional, experienced, and affordable.
            </p>
            <p
              className="lead"
            >
              Whether we like it or not, today's tax laws are so complicated that filing a relatively simple return can be confusing. It is just too easy to overlook deductions and credits to which you are entitled. Even if you use a computer software program there's no substitute for the assistance of an experienced tax professional.
            </p>
            <p
              className="lead"
            >
              We can also assist if you find yourself on the wrong side of the IRS. We're here to help you resolve your tax problems and put an end to the misery that the IRS can put you through.
            </p>
            <p
              className="lead fw-bold text-center mt-4"
            >
              Please call us today at 
              <a
                href="tel:(800) 896-4500"
              >
                (800) 896-4500
              </a>
              . Thanks for visiting!
            </p>
          </div>
          <div
            className="col-lg-1 order-lg-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
